relay-event-normalization/src/normalize/mod.rs

```diff
@@ -918,6 +941,48 @@ impl Default for LightNormalizationConfig<'_> {
             span_description_rules: Default::default(),
             geoip_lookup: Default::default(),
             enable_trimming: false,
+            dynamic_measurements_config: None,
+        }
+    }
+}
+
+/// Container for global and project level [`MeasurementsConfig`]. The purpose is to handle
+/// the merging logic.
+#[derive(Clone, Debug)]
+pub struct DynamicMeasurementsConfig<'a> {
+    project: Option<&'a MeasurementsConfig>,
+    global: Option<&'a MeasurementsConfig>,
+}
+
+impl<'a> DynamicMeasurementsConfig<'a> {
+    /// Constructor for [`DynamicMeasurementsConfig`].
+    pub fn new(
+        project: Option<&'a MeasurementsConfig>,
+        global: Option<&'a MeasurementsConfig>,
+    ) -> Self {
+        DynamicMeasurementsConfig { project, global }
+    }
+
+    /// Returns an iterator over the union of [`BuiltinMeasurementKey`]s from both the global and
+    /// project-level [`MeasurementsConfig`]s, with the elements from project config coming first.
+    pub fn builtin_measurement_keys(&'a self) -> impl Iterator<Item = &'a BuiltinMeasurementKey> {
+        let from_project = self.project.iter().flat_map(|c| &c.builtin_measurements);
+        let from_global = self.global.iter().flat_map(|c| &c.builtin_measurements);
+
+        from_project.chain(from_global).unique()
```

Comment:
The `unique` is a rather expensive operation since it needs to collect all elements. Could you instead do this with a manual `filter` function?

Basically, iterate all entries from one list, and then while iterating the ones from the second one, skip those that are in the first.

Reply:
https://github.com/getsentry/relay/pull/2415/files#diff-f09c042e62d13a4f85957ebfe2ee58d702960ac82b49955f0ef65d7001f09d50R956-R972  
 
Would this be better? 

also, a thought, maybe in the future we could cache it in the `ProjectState` struct, like when we fetch it we create the list of unique measurement keys there, so it could be re-used many times. Although then it'd have to handle updating the cache whenever there's a new global config.